### PR TITLE
Mifosx 1286

### PR DESCRIPTION
**Pull request by awaleram**

### CODE REVIEW

File: app/views/clients/clients.html [hunk 1]
```diff
@@ -3,10 +3,16 @@
...
-    <a href="#/createclient" class="btn btn-primary pull-right" has-permission='CREATE_CLIENT'><i class="icon-plus icon-white"></i>{{'label.button.createclient' | translate}}</a>
-
+    <div class="col-sm-5">
+        <input ng-autofocus="true" ng-model="filterText" ng-keyup="onFilter()" type="text" class="form-control" placeholder="{{'label.input.filterbynameaccno' | translate}}">
+    </div>
+        <form ng-submit="search()">
+            <div class="col-sm-4">
+            <input  data-ng-model="client.query" type="text" placeholder="Search Client with Name/ExternalId/Identifier/Acc.No" value="{{query}}" class="form-control"/>
+            </div>
+            <button id="query" type="search" class="btn btn-primary" has-permission='READ_CLIENT'>Search Client</button>
```
Comment (vishwasbabu) [inline app/views/clients/clients.html]: "Search Client" all text must be localized

File: app/scripts/routes.js [hunk 2]
```diff
@@ -244,6 +244,16 @@
             .when('/search/:query', {
                 templateUrl: 'views/search/glresults.html'
             })
+            .when('/clients/:query', {
```
Comment (vishwasbabu) [inline app/scripts/routes.js]: I would remove these routes too..

File: app/scripts/services/ResourceFactoryProvider.js [hunk 3]
```diff
@@ -206,6 +206,12 @@
                             isArray: true
                         }
                     }),
+                    clientSearch: defineResource(apiVer + "/client", {query: '@query'},{
```
Comment (vishwasbabu) [inline app/scripts/services/ResourceFactoryProvider.js]: clientSearch does not seem to be used anywhere

File: app/views/clients/clients.html [hunk 4]
```diff
@@ -18,6 +24,26 @@
...
+             <td class="pointer" data-ng-click="routeTo(result.entityId)">
+                 <i class="icon-stop {{result.entityStatus.code | StatusLookup}}"></i>&nbsp;{{result.entityStatus.value}}
+                 <small ng-if="result.entitySubStatus.description">[{{result.entiytSubStatus.description}}]</small></td>
+             <td class="pointer" data-ng-click="routeTo(result.entityId)"> {{result.parentName}} </td>
+         </tr>
+        </tbody>
+
+        <tbody>
+        <tr class="pointer-main" ng-repeat="result in searchResults" ng-show="result.entityType  == 'CLIENTIDENTIFIER'" >
+            <td class="pointer" data-ng-click="routeTo(result.entityId)"> {{result.parentName}} </td>
```
Comment (vishwasbabu) [inline app/views/clients/clients.html]: entity Id in the case of a ClientIdentifier refers to the entity Id of the Client Identifier itself (and not the client), so you would end up routing to the wrong client here...

File: app/views/clients/clients.html [hunk 5]
```diff
@@ -18,6 +24,26 @@
...
+            <td class="pointer" data-ng-click="routeTo(result.entityId)"> {{result.entityExternalId}} </td>
+            <td class="pointer" data-ng-click="routeTo(result.entityId)">
+                <i class="icon-stop {{result.entityStatus.code | StatusLookup}}"></i>&nbsp;{{result.entityStatus.value}}
+                <small ng-if="result.entitySubStatus.description">[{{result.entiytSubStatus.description}}]</small></td>
```
Comment (vishwasbabu) [inline app/views/clients/clients.html]: does result contain these elements ? "result.entiytSubStatus.description"

File: app/views/clients/clients.html [hunk 6]
```diff
@@ -3,10 +3,16 @@
...
+        <input ng-autofocus="true" ng-model="filterText" ng-keyup="onFilter()" type="text" class="form-control" placeholder="{{'label.input.filterbynameaccno' | translate}}">
+    </div>
+        <form ng-submit="search()">
+            <div class="col-sm-4">
+            <input  data-ng-model="client.query" type="text" placeholder="Search Client with Name/ExternalId/Identifier/Acc.No" value="{{query}}" class="form-control"/>
+            </div>
+            <button id="query" type="search" class="btn btn-primary" has-permission='READ_CLIENT'>Search Client</button>
+            <a href="#/createclient" class="btn btn-primary pull-right" has-permission='CREATE_CLIENT'><i class="icon-plus icon-white"></i>{{'label.button.createclient' | translate}}</a>
+        </form>
     <table class="table">
```
Comment (vishwasbabu) [inline app/views/clients/clients.html]: The look and feel of the new client page is very confusing

1) How would I return to the list of all clients, I would expect clearing the search box and triggering the search afresh to do so
2) Filter does not work after doing a search (As clients displayed from a search are in a different table fragment)

Comment (vishwasbabu) [inline app/views/clients/clients.html]: Same comments would apply to other html pages

Comment (awaleram) [inline app/views/clients/clients.html]: ok sir.

I will do it and I will send pull request again.

Thank you!
 Ram

Comment (vishwasbabu) [inline app/views/clients/clients.html]: @awaleram : A subsequent commit https://github.com/openMF/community-app/commit/912c11a83f1cec0914f69880e017dd1e13701038 fixes all the issues, can you please test them out at your end? You can move onto another task

Comment (awaleram) [inline app/views/clients/clients.html]: ok..sir..

thank you !

Comment (awaleram) [inline app/views/clients/clients.html]: hello sir.
I have checked merged code for jira
https://mifosforge.jira.com/browse/MIFOSX-1286. It work properly but there
is no any option to see next client.
e.g. Before merged If i have more than 100 clients , in this situation I
was see only first 15 clients on one page and for next client there was one
button at bottom for next 15 or remaining client it was working before
merged .

but right now there is no any option to see next available clients.

And in last week I have sent you two different  pull request for  jira
https://mifosforge.jira.com/browse/MIFOSX-1236 so please give mi reply for
that pull request if any suggestion or any additional requirement is there.

Thank You !
  Ram

**Pull request merged**